src/sinks/azure_blob.rs

```diff
@@ -0,0 +1,817 @@
+
+        match request {
+            Ok(_) => Ok(()),
+            Err(reason) => Err(match reason.downcast_ref::<HttpError>() {
+                Some(HttpError::UnexpectedStatusCode { received, .. }) => match *received {
+                    StatusCode::FORBIDDEN => HealthcheckError::InvalidCredentials.into(),
+                    StatusCode::NOT_FOUND => HealthcheckError::UnknownContainer {
+                        container: container_name,
+                    }
+                    .into(),
+                    status => HealthcheckError::Unknown { status }.into(),
+                },
+                _ => reason,
+            }),
+        }
+    }
+
+    pub fn create_client(&self) -> Result<Arc<ContainerClient>> {
+        let connection_string = self.connection_string.as_str();
+        let container_name = self.container_name.as_str();
+        let client =
+            StorageAccountClient::new_connection_string(new_http_client(), connection_string)?
+                .as_storage_client()
+                .as_container_client(container_name);
+
+        Ok(client)
+    }
+}
+
+impl Service<AzureBlobSinkRequest> for AzureBlobSink {
+    type Response = PutBlockBlobResponse;
+    type Error = Box<dyn std::error::Error + std::marker::Send + std::marker::Sync>;
+    type Future = BoxFuture<'static, StdResult<Self::Response, Self::Error>>;
+
+    fn poll_ready(&mut self, _cx: &mut Context<'_>) -> Poll<StdResult<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn call(&mut self, request: AzureBlobSinkRequest) -> Self::Future {
+        let client = Arc::clone(&self.client).as_blob_client(request.blob_name.as_str());
+
+        Box::pin(async move {
+            let byte_size = request.blob_data.len();
+            let blob = client
+                .put_block_blob(Bytes::from(request.blob_data))
+                .content_type(request.content_type.unwrap());
+            let blob = match request.content_encoding {
+                Some(encoding) => blob.content_encoding(encoding),
+                None => blob,
+            };
+
+            blob.execute()
+                .inspect_err(|reason| {
+                    match reason.downcast_ref::<HttpError>() {
+                        Some(HttpError::UnexpectedStatusCode { received, .. }) => {
+                            emit!(AzureBlobErrorResponse { code: *received })
+                        }
+                        _ => emit!(AzureBlobHttpError {
+                            error: reason.to_string()
+                        }),
+                    };
+                })
+                .inspect_ok(|result| {
+                    emit!(AzureBlobEventSent {
+                        request_id: result.request_id,
+                        byte_size
+                    })
+                })
+                .instrument(info_span!("request"))
+                .await
+        })
+    }
+}
+
+impl Compression {
+    pub fn content_type(&self) -> Option<&'static str> {
```

Comment:
Why are we returning an `Option` here? It seems like we always return a value.

Reply:
Hmm, I'm surprised clippy doesn't warn about this. I've definitely seen warnings about other code paths where the returned `Option` is always `Some`.

Reply:
@ArtemTrofimushkin this still needs to be updated.

Reply:
Sorry, I didn't notice that, fixed :)